src/app-bar-new.jsx

```diff
@@ -0,0 +1,271 @@
+const React = require('react');
+const Paper = require('./paper');
+const ThemeManager = require('./styles/theme-manager');
+const DefaultRawTheme = require('./styles/raw-themes/light-raw-theme');
+const StylePropable = require('./mixins/style-propable');
+
+const AppBarNew = React.createClass({
+
+	mixins: [StylePropable],
+
+	getInitialState () {
+    return {
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+    };    
+  },
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  getChildContext () {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+
+  //to update theme inside state whenever a new theme is passed down
+  //from the parent / owner using context
+  componentWillReceiveProps (nextProps, nextContext) {
+    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+    this.setState({muiTheme: newMuiTheme});
+  },
+
+	_getStyles () {
+		let spacing = this.state.muiTheme.rawTheme.spacing;
+		let appBarStyle = this.state.muiTheme.appBar;
+		return {
+			root: {
+				backgroundColor: appBarStyle.color,
+				width: '100%',
+				height: spacing.desktopKeylineIncrement,
+				paddingLeft: spacing.desktopGutter,
+				paddingRight: spacing.desktopGutter,
+				paddingTop: 20,
+				paddingBottom: 20,
+			},
+
+			navIconsGroup: {
+				maxWidth: spacing.iconSize,
+				overflow: 'hidden',
+				position: 'absolute',
+			},
+			navIcon: {
+				padding: 0,
+				width: 24,
+				height: 24,
+			},
+
+			container: {
+				position: 'absolute',
+				left: 56,
+				color: appBarStyle.textColor,
+				fontSize: 24,
+			},
+			containerElement: {
+				margin: 0,
+			},
+
+			actionIconsGroup: {
+				position: 'absolute',
+				right: 48,
+			},
+			actionIcon: {
+				padding: 0,
+				width: 24,
+				height: 24,
+				marginLeft: 24,
+				maxWidth: spacing.iconSize,
+				overflow: 'hidden',
+			},
+
+			menuIconsGroup: {
+				maxWidth: spacing.iconSize,
+				overflow: 'hidden',
+				position: 'absolute',
+				right: 0,
+			},
+			menuIcon: {
+				padding: 0,
+				width: 24,
+				height: 24,
+			},
+		};
+	},
+
+	render () {
+		return (
+      <Paper
+        rounded={false}
+        className={this.props.className}
+        style={this.mergeAndPrefix(this._getStyles().root, this.props.style)}
+        zDepth={this.props.zDepth}>
+        <div style = {{position: 'relative'}}>
+          {this._getNavIconsGroup()}
+          {this._getContainer()}
+          {this._getActionsIconsGroup()}
+          {this._getMenuIconsGroup()}
+        </div>
+      </Paper>
+    );
+	},
+
+	_getNavIconsGroup () {
```

Comment:
All these _get* methods adds a lot of overhead with the only benefit of letting the user to be able to declare children in a random order and with no grouping IMHO. Why not just make a simple component named AppBarGroup and use flexbox for layout?